File: C++/BOY OR GIRL.cpp
#include <iostream>
#include <string>

using namespace std;

void gender(string str) {
    int count = str.size();
    int x = 0;

    for (int i = 0; i < count; i++) {
        bool is_unique = true;
        for (int j = 0; j < i; j++) { // Check all previous characters
            if (str[i] == str[j]) {
                is_unique = false;
                break;
            }
        }
        if (is_unique) {
            x++;
        }
    }

    int num = x;
    if ((num % 2) == 0) {
        cout << "CHAT WITH HER!";
    } else {
        cout << "IGNORE HIM!";
    }
}

int main() {
    string name;
    cin >> name;
    gender(name);
    return 0;
}

File: Python/Boy or Girl.py
def gender(name):
    count=len(name)
    x=0
    for i in range(count):
        isUnique= True
        for j in range(i):
            if name[i]==name[j]:
                isUnique=False
                break
            j+=1  
        if(isUnique):
            x+=1
        i+=1
    
    if(x%2==0):
        print("CHAT WITH HER!")
    elif(x%2!=0):
        print("IGNORE HIM!")
name=str(input())
gender(name)

